### DISCUSSION TITLE
Feedback Needed 🙏

### DISCUSSION THREAD

**Discussion by wojciech-kulik** [wojciech-kulik/xcodebuild.nvim#44] [Announcements]
If you are a user of xcodebuild.nvim plugin, please share your feedback here.  It's a weird feeling when the repository has over 200 stars, but I know nothing if this plugin works or not on other machines :D. It was only tested on my machine and I have no idea how it behaves on other setups.

I'm an iOS developer so I got used to analytics, but here is like a black hole, not sure if people install it, it doesn't work and they dump it, or if it works great and that's why there are no issues 😂 .

I'm curious if everything works, if you managed to try out all the features or if you are missing something. So If you find this discussion, please share your thoughts 🙏 

PS. It would be helpful if you could share your environment very briefly like: macOS version, CPU type, project platform (iPhone, iPad, macOS) + version, and if you tested it on physical devices using this plugin, so that I know at least which configurations are confirmed to be working :D

**Comment by PangMo5** [comment]
Hi first of all thank you for the awesome plugin.

I came across your plugin about 2 weeks ago and have been diving into the world of neovim and am getting used to it.

With the help of this plugin and other plugins and nvim settings, I'm actually using neovim in my company and personal projects in a very natural way, more than using Xcode.

I also really like the recent updates, like nvim-tree xcodeproj support and debug support for physical devices, which are very cool!

The only thing is that I seem to have a problem with project not building in certain environments.
I'm using two environments, alacritty (with zellij, fish) and neovide. If I run the nvim or neovide command in the root of the project in the shell to set the environment, the build works fine.
But when I run neovide directly as an app and enter the recent project(same root) through dashboard.nvim, the UI of Lualine etc. looks normal, but when I try to build it, it only goes up to seconds count and continues indefinitely.
XcodebuildCancel command doesn't work in this situation either.

My environment is macOS 14.3, M1 Pro, iOS project, debug with iOS 17.3.1 physical device, iPhone 15 Pro (17.2) Simulator

**Reply by wojciech-kulik** [reply]
Thank you very much for the feedback! I'm not familiar with neovide, however, the plugin requires cwd to be set in the project root. You can try checking `:lua print(vim.fn.getcwd())` and see if it's set correctly. If you still have problems with it you can open an issue.

**Reply by PangMo5** [reply]
Thanks for the answer.
But the `:lua print(vim.fn.getcwd())` output in both situations looks the same.

I tried creating a new project just in case, but I'm seeing the same issue.

Even when using nvim (not neovide)
run `nvim {project-root}` shell command from the parent root of the project, the same problem seems to occur.

If it's not just me, I'll open an issue.

#### When running as a shell command from the project root
![SCR-20240221-wxj](https://github.com/wojciech-kulik/xcodebuild.nvim/assets/16532526/48e6fd97-dc43-4bfc-a0c9-c350af6c08fc)

#### When running as Dashboard.nvim, etc.
![SCR-20240221-wyh](https://github.com/wojciech-kulik/xcodebuild.nvim/assets/16532526/61036ff9-83e2-468c-be1e-573386f289a4)

**Reply by wojciech-kulik** [reply]
Please convert it into a issue anyway, I will try verifying it :)

**Reply by wojciech-kulik** [reply]
Resolved: #45

**Comment by Almaz5200** [comment]
Hi!

First of all, many thanks to you for this project, I am finally able to combine my favourite editor with my actual job :)
I integrate this plugin with my [LazyVim](https://www.lazyvim.org/) setup, which works great for the most part.
There is a few points I would love to raise that could lead to this wonderful project being even better:

1. Target switching. Currently I do it with `XcodebuildSelectScheme`, but it does not call `xcode-build-server` to change buildServer.json, which makes it difficult to switch targets at times, which I have to do pretty frequently due to the nature of my project

2. Contribution guidelines. Unfortunately I have no experience developing neovim plugins, because of that I don't even know where to start, even how to use some "local" version of it instead of pulling it from github. I've had a few moments when I would want to add certain features that I feel should not be too difficult, but that uncertainty stops me, which I feel like could be an issue for a lot of developers. This is why it would be great to have a little "getting started" page for contributes which outlines how to hook local version up to neovim and how it works broadly.

3. Tuist integration. We use tuist to manage our project, which has it's own set of libraries for configuration. Their suggested flow is "run `tuist edit` which creates temporary xcode project with those libraries hooked up". While it should be possible to reconfigure xcode-build-server to look at that xcproject, it seems like too much work, and since usually all I want is few simple changes I usually either do it without LSP working or use Xcode.
I am not sure if this can be done in any simple way, but it would be great to have LSP working properly in those files as well
Kinda feels like it boils down to being able to hook up two projects at the same time

My setup is M1 Pro MacBook, iOS 15+ only project that requires lots of Target switching, I only debug on simulators for now, but looking to explore new features with running on physical devices

**Reply by wojciech-kulik** [reply]
Thank you for the feedback! 🍻  I will definitely look into the points you mentioned! For sure, contribution guidelines should be doable.

**Reply by wojciech-kulik** [reply]
@Almaz5200 as promised, I prepared the repository for contribution. I added tools to ensure quality standards, documented the WHOLE code base (almost 4000 lines of help 😱 ), I added [CONTRIBUTING.md](https://github.com/wojciech-kulik/xcodebuild.nvim/blob/main/CONTRIBUTING.md), and I set up GitHub actions to ensure quality checks for PRs. Enjoy!

**Reply by Almaz5200** [reply]
Thank you, I appreciate that a lot! In fact, my first pull request is coming soon, I've already noticed contribution guidelines about an hour ago and started working on it right away!
